From 7115888a573c734d00bfb6eb491228b70981442d Mon Sep 17 00:00:00 2001
From: gbhakteX <gurudattax.bhakte@intel.com>
Date: Wed, 26 Sep 2012 19:33:00 +0530
Subject: [PORT FROM MAIN]Audio-MW: Not able to play AAC-LD files through music app.

BZ: 56637

Currently MDP Intel optimized AAC decoder suite doesn't have AAC-LD
decoding capability.
This fix provides capability to invoke Google AAC decoder for AAC-LD
decoding.
Signed-off-by: gbhakteX <gurudattax.bhakte@intel.com>

Change-Id: I928fcfa56969df384aff326b18078703430981f3
Orig-Change-Id: I547bd267f50bd56f724953177de32129d20a3c3c
---
 media/libstagefright/MPEG4Extractor.cpp |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/media/libstagefright/MPEG4Extractor.cpp b/media/libstagefright/MPEG4Extractor.cpp
index 25d4610..a32473f 100644
--- a/media/libstagefright/MPEG4Extractor.cpp
+++ b/media/libstagefright/MPEG4Extractor.cpp
@@ -38,6 +38,7 @@
 #include <media/stagefright/MediaSource.h>
 #include <media/stagefright/MetaData.h>
 #include <utils/String8.h>
+#include <media/openmax/OMX_Audio.h>
 
 namespace android {
 
@@ -2333,10 +2334,12 @@ status_t MPEG4Extractor::updateAudioTrackInfoFromESDS_MPEG4Audio(
 
     if (objectType == 31) {  // AAC-ELD => additional 6 bits
         objectType = 32 + br.getBits(6);
+    }
 #ifdef USE_INTEL_MDP
+    if ((objectType == OMX_AUDIO_AACObjectLD) || (objectType >= 31)) {
         mLastTrack->meta->setCString(kKeyMIMEType, MEDIA_MIMETYPE_AUDIO_AAC_EXTENDED);
-#endif
     }
+#endif
     mLastTrack->meta->setInt32(kKeyAACProfile, objectType);
 
     uint32_t freqIndex = br.getBits(4);
-- 
1.7.4.1

